src/dmd/backend/cgobj.d

```diff
@@ -99,10 +95,25 @@ else
     pragma(printf) void error(Loc loc, const(char)* format, ...);
 }
 
-int obj_namestring(char *p,const(char)* name);
-
-static if (TARGET_WINDOS)
+version (Windows)
 {
+    extern(C) char* strupr(char*);
+}
+version (Posix)
+{
+    extern(C) char* strupr(char* s)
```

Comment:
Use dmd.root.port.strupr?

Reply:
I keep the backend separate from the front end.

Reply:
You are already using dmd.root.hash and dmd.root.longdouble in the backend, and port is a leaf module (it only depends on druntime).  So I don't really see that as an acceptable argument.

Reply:
There are other redundancies between the back and front ends, but recall the same backend is used for DMC++, and entangling it with the D frontend will cause me all sorts of headaches.

Reply:
`version (MARS)` ought to cover the dmd/dmc divide then.